marlowe-playground-client/src/Wallet.purs

```diff
@@ -0,0 +1,1301 @@
+                , a [ onClick $ const $ Just $ ChangeHelpContext TransactionComposerHelp ] [ img [ src infoIcon, alt "info book icon" ] ]
+                ]
+            ]
+          <> [ div [ classes [ ClassName "transaction-composer", ClassName "composer" ] ]
+                ( if hasTransactions then
+                    [ transactionComposer state
+                    , div [ class_ (ClassName "transaction-btns") ]
+                        [ ul [ classes [ ClassName "demo-list", aHorizontal ] ]
+                            [ li [ class_ (ClassName "apply-transaction-button") ]
+                                [ button
+                                    [ onClick $ Just <<< const ApplyTransaction
+                                    , enabled hasTransactions
+                                    , class_ (Classes.disabled $ not hasTransactions)
+                                    ]
+                                    [ text "Apply" ]
+                                ]
+                            ]
+                        ]
+                    ]
+                  else
+                    [ text "no transactions" ]
+                )
+            ]
+      )
+  ]
+  where
+  pendingInputs = fromMaybe mempty (state ^? (_currentLoadedMarloweState <<< _pendingInputs))
+
+  possibleActions =
+    fromMaybe mempty
+      $ state
+      ^? ( _currentLoadedMarloweState
+            <<< _possibleActions
+            <<< to (Map.filterKeys walletKeys)
+            <<< to Map.values
+            <<< to (map Map.toUnfoldable)
+            <<< to fold
+        )
+
+  walletKeys (Just party) =
+    let
+      parties = fromMaybe mempty $ state ^? (_walletLoadedContract <<< _Just <<< _parties)
+
+      walletParties = case state ^? (_openWallet <<< _Just) of
+        Just wallet -> Set.fromFoldable $ Map.keys $ Map.filterWithKey (\_ owner -> owner == (wallet ^. _name)) parties
+        Nothing -> mempty
+    in
+      Set.member party walletParties
+
+  walletKeys _ = false
+
+  hasTransactions = fromMaybe false $ state ^? (_currentLoadedMarloweState <<< _pendingInputs <<< to Array.null <<< to not)
+
+  renderAction :: (Tuple ActionInputId ActionInput) -> HTML p Action
+  renderAction (Tuple actionInputId actionInput) = inputItem true "" actionInput
+
+renderMyRoles :: forall p. PubKey -> State -> HTML p Action
+renderMyRoles me state =
+  div [ classes [ ClassName "centre-element" ] ]
+    [ h6 [ classes [ ClassName "parties-heading", noMargins ] ]
+        [ small [ classes [ textSecondaryColor, bold, uppercase ] ] [ text "My Roles" ] ]
+    , Keyed.ul [] (map (displayMyRole me wallets) myRoles)
+    ]
+  where
+  wallets = map snd $ Map.toUnfoldable $ state ^. _wallets
+
+  myRoles =
+    Array.fromFoldable
+      $ Set.mapMaybe
+          ( \(Tuple contract party) -> do
+              name <- getRoleName party
+              pure (Tuple contract name)
+          )
+      $ Map.keys
+      $ Map.filterWithKey (\_ v -> v == me)
```

Comment:
Why are you using `filterWithKey` only to ignore the key? 🤔

Reply:
your guess is as good as mine